core/server/api/schedules.js

```diff
@@ -0,0 +1,61 @@
+var _ = require('lodash'),
+    Promise = require('bluebird'),
+    moment = require('moment'),
+    config = require('../config'),
+    pipeline = require(config.paths.corePath + '/server/utils/pipeline'),
+    dataProvider = require(config.paths.corePath + '/server/models'),
+    i18n = require(config.paths.corePath + '/server/i18n'),
+    errors = require(config.paths.corePath + '/server/errors'),
+    utils = require('./utils');
+
+/**
+ * publish a scheduled post
+ */
+exports.publishPost = function publishPost(options) {
+    options = options || {};
+
+    // CASE: only the scheduler client is allowed to publish (hardcoded because of missing client permission system)
+    if (!options.context || !options.context.client || options.context.client !== 'ghost-scheduler') {
+        return Promise.reject(new errors.NoPermissionError(i18n.t('errors.permissions.noPermissionToAction')));
+    }
+
+    return pipeline([
+        utils.validate('posts', {opts: utils.idDefaultOptions}),
+        function (cleanOptions) {
+            return dataProvider.Post.edit({status: 'published'}, _.omit(cleanOptions, 'data'))
```

Comment:
i will add the X-Cache-Invalidation. i've added a task.
the 404 issue is the same we discussed already here: 
https://github.com/TryGhost/Ghost/pull/6861/files#r66493705 - we will add a check!